src/oscar/apps/order/abstract_models.py

```diff
@@ -223,24 +223,24 @@ def shipping_tax(self):
 
     @property
     def shipping_status(self):
-        events = self.shipping_events.all()
+        """Return the last complete shipping event for this order."""
+        events = self.shipping_events.order_by('-date_created').all()
```

Comment:
Could you add sorting by primary key as well, and leave a comment to explain why?
e.g. `# As safeguard against identical time stamps, we also sort by primary key. It's not recommended to rely on this behaviour, but in practice reasonably safe if PKs are not manually set.`
I confirmed this behaviour with @aaugustin.